cpp/dolfin/fem/assemble_matrix_impl.cpp

```diff
@@ -225,6 +228,7 @@ void fem::impl::assemble_exterior_facets(
     // Get local index of facet with respect to the cell
     const int local_facet = cell.index(facet);
     const int orient = 0;
+    const int perm = cell.facet_permutation(facet);
```

Comment:
`facet_permutation()` is quite heavy in logic and contains loads of ifs and branches. I think you should rather precompute `const int perm` for possible connectivities and store somewhere. Assemblers should rather just access the contiguous data, not do such computations. I am sure @garth-wells would agree.

Reply:
Agreed, I'll work on getting this and `Mesh::get_entity_reflections` (https://github.com/FEniCS/dolfinx/blob/b6fee46adada0eaedae7c178871dc61fcb4771be/cpp/dolfinx/mesh/Mesh.cpp#L398-L434) precomputed